src/System.IO.Packaging/src/System/IO/Packaging/IgnoreFlushAndCloseStream.cs

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+//------------------------------------------------------------------------------
+//  File:           IgnoreCloseAndFlushStream.cs
+//
+//  Description:    The class is used to wrap a given stream in a way that the Flush
+//                  and Close calls to the stream are Ignored. This stream class has been
+//                  created specifically for perf improvements for the ZipPackage.
+//------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+
+namespace System.IO.Packaging
+{
+    /// <summary>
+    /// This class ignores all calls to Flush() and Close() methods
+    /// depending on whether the IgnoreFlushAndClose property is set to true
+    /// or false.
+    /// </summary>
+    internal sealed class IgnoreFlushAndCloseStream : Stream
+    {
+        #region Constructor
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="stream"></param>
+        internal IgnoreFlushAndCloseStream(Stream stream)
+        {
+            if (stream == null)
+                throw new ArgumentNullException("stream");
+
+            _stream = stream;
+        }
+
+        #endregion Constructor
+
+        #region Properties
+
+        /// <summary>
+        /// Member of the abstract Stream class
+        /// </summary>
+        /// <value>Bool, true if the stream can be read from, else false</value>
+        public override bool CanRead
+        {
+            get
+            {
+                if (_disposed)
+                    return false;
+                else
+                    return _stream.CanRead;
```

Comment:
Again, maybe a style thing, but I'd have written the whole body as:
```C#
get { return !_disposed && _stream.CanRead; }
```
I'll stop commenting on these.